server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -586,6 +586,8 @@ protected Node(
                 pluginCircuitBreakers,
                 settingsModule.getClusterSettings()
             );
+            // File cache will be initialized by the node once circuit breakers are in place.
+            nodeEnvironment.initializeFileCache(settings, circuitBreakerService.getBreaker(CircuitBreaker.REQUEST));
```

Comment:
The initialization had to be moved out here due to the `CircuitBreakerService` initialization. I realize this is not ideal, but the sequence of initialization forces this out. 
Please chime in if you have a better approach.

Reply:
Digging into this a bit, it seems to me that NodeEnvironment should not hold a `fileCache` instance and instead should just hold information about the _path_ of the file cache, as NodeEnvironment is defined as "A component that holds all data paths for a single node."

I think this can be refactored as follows:
- Remove the actual file cache instance from NodeEnvironment
- Create the filecache instance in Node.java where are you currently calling `nodeEnvironment.initializeFileCache`
- In Node.java, create a FileCacheCleaner instance using the FileCache instance (there is nothing per-index about the cleaner, so I think only a single instance is file). Pass that FileCacheCleaner instance to IndicesService to wire up as a listener.
- Profit!

What do you think?